File: scrimage-core/src/main/java/com/sksamuel/scrimage/canvas/drawables/Text.java
package com.sksamuel.scrimage.canvas.drawables;

import com.sksamuel.scrimage.canvas.Drawable;
import com.sksamuel.scrimage.canvas.GraphicsContext;
import com.sksamuel.scrimage.graphics.RichGraphics2D;

public class Text implements Drawable {

    private final String text;
    private final int x;
    private final int y;
    private final GraphicsContext context;

    public Text(String text, int x, int y, GraphicsContext context) {
        this.text = text;
        this.x = x;
        this.y = y;
        this.context = context;
    }

    @Override
    public void draw(RichGraphics2D g2) {
        // Canvas.draw / drawInPlace already invoke `d.context().configure(g2)`
        // before calling d.draw(g2) — the same way it does for every other
        // Drawable. This class used to call configure() a second time here,
        // which was a no-op for overwriting setters (setColor/setFont/...) but
        // applied additive operations (translate/rotate/transform) twice.
        g2.drawString(text, x, y);
    }

    @Override
    public GraphicsContext context() {
        return context;
    }
}

File: scrimage-core/src/main/java/com/sksamuel/scrimage/canvas/drawables/Text.scala
package com.sksamuel.scrimage.canvas.drawables

import java.awt.Graphics2D

import com.sksamuel.scrimage.canvas.{Drawable, GraphicsContext}

case class Text(text: String, x: Int, y: Int, context: GraphicsContext) extends Drawable {
  def draw(g2: Graphics2D): Unit = {
    context.configure(g2)
    g2.drawString(text, x, y)
  }
}

object DrawableString {
  @deprecated("Use Text", "3.0.0")
  def apply(text: String, x: Int, y: Int, context: GraphicsContext): Text = Text(text, x, y, context)
}